### PR TITLE
Fix some fetcher test

### PR DESCRIPTION
**Pull request by DominikVoigt**
This PR fixes some of the failing fetcher tests specified in #6369.
Namely:
* [x]  Jstor: Result differs
* [x]  CiteSeer: Result differs
* [x]  TitleFetcher: Change format
* [x]  Disable CompositeFetcherTest on CI

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Review by calixtus — APPROVED** [review]

File: src/test/java/org/jabref/logic/importer/fetcher/CompositeSearchBasedFetcherTest.java [hunk 1]
```diff
@@ -28,6 +29,7 @@
 import static org.mockito.Mockito.when;
 
 @FetcherTest
+@Disabled("Produces to many requests on CI")
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/importer/fetcher/CompositeSearchBasedFetcherTest.java:32]:
```suggestion
@DisabledOnCIServer("Produces to many requests on CI")
```

**Review by Siedlerchr — APPROVED** [review]
Just one minor annoation change

**Comment by calixtus** [comment]
/home/runner/work/jabref/jabref/src/test/java/org/jabref/logic/importer/fetcher/CompositeSearchBasedFetcherTest.java:32: error: cannot find symbol
@DisabledOnCIServer("Produces to many requests on CI")
 ^
  symbol: class DisabledOnCIServer

**Comment by DominikVoigt** [comment]
Fixed! 😄

**Pull request merged**